File: Army(c++)/Warriors/Priest.cpp
#include "Priest.hpp"
#include "../State/State.hpp"

Priest::Priest(const std::string& name,  int hp, int mana, int dmg) 
    : SpellCaster(name, hp, mana, dmg, new SoldierState(this))
    , warrior(WarriorID::PRIEST), magician(MagicianID::CHURCH_WORKER) {}

Priest::~Priest() {}

WarriorID Priest::getWarriorID() const {
    return this->warrior;
}

void Priest::setWarriorID(WarriorID newID) {
    this->warrior = newID;
}

MagicianID Priest::getMagicianID() const {
    return this->magician;
}

File: Army(java)/Warriors/Priest.java
package Warriors;

import EnumID.*;
import Basic.*;

public class Priest extends SpellCaster {
    private WarriorID warrior;
    private MagicianID magician;

    public Priest(String name,  int hp, int mana, int dmg) {
        super(name, hp, mana, dmg);
        this.warrior = WarriorID.PRIEST;
        this.magician = MagicianID.CHURCH_WORKER;
    }

    public WarriorID getWarriorID() {
        return warrior;
    }

    public void setWarriorID(WarriorID newID) {
        warrior = newID;
    }

    public MagicianID getMagicianID() {
        return magician;
    }
}